cmake/projects/Boost/scripts/patched_boostrap.bat.in

```diff
@@ -1,4 +1,4 @@
-call "@HUNTER_MSVC_VCVARSALL@" @HUNTER_MSVC_ARCH@
+call "@HUNTER_MSVC_VCVARSALL@" x86
```

Comment:
Why is x86 hardcoded here?

Reply:
As far as I can tell Visual Studio is not provided as a native ARM64 binary (I may be wrong about this!), only x86 and x64 versions of the compilers are available. Hunter uses an x86 compiler for ARMv7 (i.e. it chooses `x86_arm` for ARMv7), and now uses an x86 compiler for ARM64.

The vcvarsall script here is setting up the build to run the bootstrap script which builds `b2`, the Boost build system. We need the build system to be build for the HOST rather than the TARGET system so that we can run it to build the rest of Boost.

I hardcoded `x86` as that will work in both x86 and x64 situations. This could probably be improved throughout hunter MSVC support as it just picks `x86` compiler and tools in `hunter_setup_msvc.cmake`, when it could try to determine from CMake that the user had chosen to use the `x64` tools.

Reply:
Won't this break x64-native builds? Toolchain `vs-16-2019-win64.cmake` with `-A x64` for example.

Reply:
This will cause `b2` to always be built as `x86` yes. The issue here is that `HUNTER_MSVC_ARCH` is the arch we want the resulting library to be, not neccesarily an arch that can run locally (to build and run `b2`).

I do not believe that anything in this bootstrap script affects what happens when building the actual libraries, but I would appreciate if someone could independantly confirm that. :)

Reply:
I think 64bit native build should be able to run the x86 binary?

Reply:
Maybe, but that doesn't seem right if a Hunter developer expects to have an x64 build all the way through. Perhaps there's a better way to do this?

Reply:
OK, I have broken out part of this change and will consider how to do this better.